### PR TITLE
Update disabled job check with forward compatibility

### PR DESCRIPTION
**Pull request by timja**
see https://github.com/jenkinsci/jenkins/pull/6485

Alternative to https://github.com/jenkinsci/workflow-job-plugin/pull/261 which can be merged ahead of time

cc @janfaracik

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by basil — COMMENTED** [review]
It does not appear this change was tested with `jenkins.version` set to an incremental build of https://github.com/jenkinsci/jenkins/pull/6485:

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 11.93 s <<< FAILURE! - in org.jenkinsci.plugins.workflow.job.WorkflowJobTest
[ERROR] org.jenkinsci.plugins.workflow.job.WorkflowJobTest.disabled  Time elapsed: 11.879 s  <<< ERROR!
com.gargoylesoftware.htmlunit.ElementNotFoundException: elementName=[input] attributeName=[name] attributeValue=[disable]
        at com.gargoylesoftware.htmlunit.html.HtmlForm.getInputByName(HtmlForm.java:620)
        at org.jenkinsci.plugins.workflow.job.WorkflowJobTest.disabled(WorkflowJobTest.java:96)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:606)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by timja** [comment]
There’s no incremental at least at the time but yes aware will fix that this morning

**Review by basil — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowJobTest.java [hunk 1]
```diff
@@ -93,9 +94,16 @@ public void addAction() throws Exception {
...
         HtmlForm form = wc.getPage(p, "configure").getFormByName("config");
-        HtmlCheckBoxInput checkbox = form.getInputByName("disable");
-        assertTrue(checkbox.isChecked());
-        checkbox.setChecked(false);
+        DomNode domNode = form.querySelector("input[name='enable']");
+        if (domNode != null) {
+            HtmlCheckBoxInput checkbox = (HtmlCheckBoxInput) domNode;
+            assertFalse(checkbox.isChecked());
+            checkbox.setChecked(true);
+        } else {
```
Comment (timja) [inline src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowJobTest.java:102]:
```suggestion
        } else {
        // TODO remove compatibility when baseline includes https://github.com/jenkinsci/jenkins/pull/6485
```

File: src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowJobTest.java [hunk 2]
```diff
@@ -93,9 +94,17 @@ public void addAction() throws Exception {
...
+        // TODO remove compatibility when baseline includes https://github.com/jenkinsci/jenkins/pull/6485
```
Comment (timja) [inline src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowJobTest.java:103]:
```suggestion
          // TODO remove compatibility when baseline includes https://github.com/jenkinsci/jenkins/pull/6485
```

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java [hunk 3]
```diff
@@ -188,7 +188,12 @@ public void setDefinition(FlowDefinition definition) {
         } else {
             quietPeriod = null;
         }
-        makeDisabled(json.optBoolean("disable"));
+        // TODO remove compatibility when baseline includes https://github.com/jenkinsci/jenkins/pull/6485
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java:191]: Could you clarify in the comment which side is the compatibility side vs. the new side?

File: src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowJobTest.java [hunk 4]
```diff
@@ -93,9 +94,17 @@ public void addAction() throws Exception {
...
-        HtmlCheckBoxInput checkbox = form.getInputByName("disable");
-        assertTrue(checkbox.isChecked());
-        checkbox.setChecked(false);
+        DomNode domNode = form.querySelector("input[name='enable']");
+        if (domNode != null) {
+            HtmlCheckBoxInput checkbox = (HtmlCheckBoxInput) domNode;
+            assertFalse(checkbox.isChecked());
+            checkbox.setChecked(true);
+        } else {
+            // TODO remove compatibility when baseline includes https://github.com/jenkinsci/jenkins/pull/6485
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowJobTest.java:103]: Also here—not clear to a reader without digging into the upstream PR which version is old and which is new.

Comment (basil) [reply src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowJobTest.java]: > not clear to a reader without digging into the upstream PR which version is old and which is new.

You mean a TODO comment lacked sufficient context for someone not already familiar with the problem? Yes, just as in https://github.com/jenkinsci/jenkins-test-harness/pull/428#discussion_r863019143. In that PR, you did not respond to the review feedback.

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java (hunk 3 above)
Comment (timja) [reply src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java]: I wouldn't worry too much about the comment, there's a PR to update the baseline to take advantage of the new APIs https://github.com/jenkinsci/workflow-job-plugin/pull/261 the TODO will be resolved there.

**Comment by timja** [comment]
@jglick or @jenkinsci/workflow-job-plugin-developers could you take a look please?

**Review by car-roll — APPROVED** [review]

**Comment by timja** [comment]
@car-roll could you click the merge button please?

**Pull request merged**